glean-core/src/metrics/labeled.rs

```diff
@@ -16,7 +16,26 @@ const MAX_LABELS: usize = 16;
 const OTHER_LABEL: &str = "__other__";
 const MAX_LABEL_LENGTH: usize = 61;
 lazy_static! {
-    static ref LABEL_REGEX: Regex = Regex::new("^[a-z_][a-z0-9_]{0,29}$").unwrap();
+    /// This regex is used for matching against labels and should allow for dots, underscores,
+    /// and/or hyphens. Labels are also limited to starting with either a letter or an
+    /// underscore character.
+    /// Some examples of good and bad labels:
+    ///
+    /// Good:
```

Comment:
This will render as a single line in the rendered doc (in theory, it's a private item, so by default it's not even shown)

Reply:
Alternative: make it a list and avoid markdown swallowing underscores.

```
* `this.is.fine`
```